novawallet/Modules/AssetPriceChart/Factory/AssetPriceChartViewModelFactory.swift

```diff
@@ -184,8 +185,19 @@ extension AssetPriceChartViewModelFactory: AssetPriceChartViewModelFactoryProtoc
             selectedPeriod: params.selectedPeriod
         )
 
+        var entries: [PriceHistoryItem] = []
+
+        if let allEntries = params.entries {
+            if allEntries.count > params.availablePoints {
+                entries = allEntries
+                    .distribute(intoChunks: params.availablePoints)
+                    .compactMap(\.first)
```

Comment:
In that case we might miss the latest price. What do you think is we include first and last prices anyway and select remaining from the distribution?